### PR TITLE
Implement Form Localisation

### PR DESCRIPTION
**Pull request by Kiriox94**
Implementation of the ability to translate pokemon form, not being very advanced in the game I wasn't able to test if it worked well.

### CODE REVIEW

File: src/locales/en/pokemon-form.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    "koraidonGlidingBuild": "Gliding Build",
+    "miraidonUltimateMode": "Ultimate Mode",
+    "miraidonLowPowerMode": "Low-Power Mode",
+    "miraidonDriveMode": "Drive Mode",
+    "miraidonAquaticMode": "Aquatic Mode",
+    "miraidonGlideMode": "Glide Mode",
+    "poltchageistCounterfeit": "Counterfeit Form",
+    "poltchageistArtisan": "Artisan Form",
+    "sinistchaUnremarkable": "Unremarkable Form",
+    "sinisrchaMasterpiece": "Masterpiece Form",
```
Comment (Madmadness65) [inline src/locales/en/pokemon-form.ts:219]: Minor nitpick, all instances of Sinistcha's Masterpiece Form has the typo `sinisrcha`.

File: src/plugins/i18n.ts [hunk 2]
```diff
@@ -131,6 +137,7 @@ export function initI18n(): void {
         commandUiHandler: enCommandUiHandler,
         fightUiHandler: enFightUiHandler,
         tutorial: enTutorial,
+        pokemonForm: enPokemonForm,
```
Comment (Flashfyre) [inline src/plugins/i18n.ts:140]: Everywhere you order this, it should go after Pokemon, not at the end.

File: src/locales/de/pokemon-form.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    "alcremieSaltedCream": "Salted Cream",
+    "alcremieRubySwirl": "Ruby Swirl",
+    "alcremieCaramelSwirl": "Caramel Swirl",
+    "alcremieRainbowSwirl": "Rainbow Swirl",
+    "eiscueIceFace": "Ice Face",
+    "eiscueNoIce": "No Ice",
+    "morpekoFullBelly": "Full Belly Mode",
+    "morpekoHangry": "Hangry Mode",
+    // Legendary Heroes is Zacian ans Zamazenta
+    "LegendaryHeroesHero": "Hero of Many Battles",
```
Comment (Flashfyre) [inline src/locales/de/pokemon-form.ts:181]: L should not be capital. Also, fix typo in comment

**Review by Flashfyre — CHANGES_REQUESTED** [review]

**Comment by Adri1** [comment]
Are these form names supposed to be paired with Pokémon names, or just displayed by themselves? Because if paired with Pokémon names, I would like to adress a potential word order issue.
In English, the name of the form is always **before** the name of the Pokémon.
But in languages like French, it's mostly placed **after** the name of the Pokémon, with a few exceptions like "Méga", "Primo"or "Ultra", always placed before.

Furthermore, if they are supposed to be paired with Pokémon names AND be used by themselves, so I'd like to point out that some form comes with extra features when paired with a Pokémon name. For example "Méga", "Primo" and "Ultra" comes with an extra dash instead of a space when pair with a Pokémon name like :
-Méga-Steelix
-Primo-Groudon
-Ultra-Necrozma
So either in the translation we write "Méga" and it would be incorret when paired with a Pokémon name, or we write "Méga-" and would be incorrect too if it is not

**Comment by Kiriox94** [comment]
Yes, the display order is a bit awkward, I'll probably do another pull request to fix it, and as for the dash, do you think it's really annoying? Personally, once I'm in the game, I don't pay attention to this kind of detail, but if it really bothers you, I can try to find a solution.

**Comment by Adri1** [comment]
> Yes, the display order is a bit awkward, I'll probably do another pull request to fix it, and as for the dash, do you think it's really annoying? Personally, once I'm in the game, I don't pay attention to this kind of detail, but if it really bothers you, I can try to find a solution.

About the dash, it's not that much even if it's their official orthography, but idk the importance of it the other languages, like German, also using dashes in this context like in French.

Also if someday we have a Japanese translation, this kind of issue might come back in some ways, as for example they completly glue the word "Mega" (メガ) to the name of the Pokémon
And in some cases they add a particle when the name of the form is glued to a Pokémon, as when they talk about a regional form of a Pokémon between the name of the form and the name of the Pokémon, kida like the dashes I'm taking about.
(I know regional forms aren't part of theses files, but it's just to give you an example of somthing that may occur)

**Comment by CodeTappert** [comment]
> About the dash, not that much even if it's their official orthography, but idk the importance of it the other languages, like German, also using dashes in the context of Mega, Ultra and Primal, like in French.
> 

German also uses the Dash for Mega Pokemon. 

For example `Mega-Rayquaza`. Same for `Proto-Groudon` and `Proto-Kyogre`. As well as `Ultra-Necrozma`.

Some form are added in () at the end. Like `Giratina (Urform)`

**Comment by Adri1** [comment]
Any news about this?

**Comment by Kiriox94** [comment]
I'm sorry I've stopped working on it a bit, the simplest thing would be that I have all the different ways to display a form in all supported languages, but I don't know how to do that

**Comment by Kiriox94** [comment]
Oh, sorry I made a wrong manipe it closed the pr, I'll reopen one when the requested changes are ready.

**Pull request closed without merging**